### PR TITLE
Remove native `chmod(2)` and `stat(2)` support

### PR DESCRIPTION
**Pull request by basil**
A step on the road to eventually removing JNR (and therefore ASM) from Jenkins core. These are the only two usages in core (although there are more usages in plugins), so we might as well get rid of them now. If this change is merged, then I'll look into weaning plugins off of JNR and ASM and then eventually removing JNR and ASM from core. I'll also update [the documentation](https://www.jenkins.io/doc/book/managing/system-properties/) to mark the property as obsolete.

### Proposed changelog entries

Removed: Support for native (JNR) `chmod(2)` and `stat(2)` implementations (as opposed to NIO) via the `hudson.Util.useNativeChmodAndMode` system property has been removed. This system property no longer has any effect.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
What's the reason this was used before? Limitations in Java API of the time?

**Comment by basil** [comment]
> What's the reason this was used before? Limitations in Java API of the time?

Yes, NIO wasn't added until Java 7.

**Review by timja — APPROVED** [review]

File: core/src/main/java/hudson/Util.java
Comment (daniel-beck) [inline core/src/main/java/hudson/Util.java:1815]: Is this still relevant? (I think we can probably go without, but looks like it deserves some investigation if you haven't yet.)

Comment (basil) [reply core/src/main/java/hudson/Util.java:1815]: Quoth Devin in [JENKINS-46725 (comment)](https://issues.jenkins.io/browse/JENKINS-46725?focusedCommentId=341691&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-341691):

> Yes, `hudson.Util#NATIVE_CHMOD_MODE` should be able to be removed. It was added to Jenkins 2.93 as part of [JENKINS-36088](https://issues.jenkins.io/browse/JENKINS-36088) to help remove unnecessary usages of native code. There was a small compatibility concern with the Java vs native implementations so I defaulted to the Java impl and added a system property to enable the native implementation, but I have not seen any reports from users that needed to enable the native implementation so I do not anticipate any issues with removing the option.

**Review by res0nance — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. After ~24 hours, we will merge it if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
